ios/Video/AudioSessionManager.swift

```diff
@@ -0,0 +1,180 @@
+        }
+    }
+
+    private func getCategory(silentSwitchObey: Bool, earpiece: Bool, pip: Bool, needsPlayback: Bool) -> AVAudioSession.Category {
+        if needsPlayback {
+            if earpiece {
+                RCTLogWarn(
+                    """
+                    You can't set \"audioOutput\"=\"earpiece\" and \"ignoreSilentSwitch\"=\"obey\"
+                    at the same time (in same or different components) - skipping those props
+                    """
+                )
+                return .playback
+            }
+
+            if silentSwitchObey {
+                RCTLogWarn(
+                    """
+                    You can't use \"playInBackground or \"notificationControls with \"ignoreSilentSwitch\"=\"obey\"
+                    at the same time (in same or different components) - skipping \"ignoreSilentSwitch\" prop
+                    """
+                )
+                return .playback
+            }
+        }
+
+        if silentSwitchObey {
+            if earpiece {
+                RCTLogWarn(
+                    """
+                    You can't set \"audioOutput\"=\"earpiece\" and \"ignoreSilentSwitch\"=\"obey\"
+                    at the same time (in same or different components) - skipping those props
+                    """
+                )
+                return .playback
+            }
+
+            if pip {
+                RCTLogWarn(
+                    """
+                    You use \"pictureInPicture\"=\"true\" and \"ignoreSilentSwitch\"=\"obey\"
+                    at the same time (in same or different components) - skipping those props
+                    """
+                )
+                return .playback
+            }
+
+            return .ambient
+        }
+
+        return earpiece ? .playAndRecord : .playback
+    }
+
+    func updateAudioSessionCategory() {
+        let audioSession = AVAudioSession()
+        var options: AVAudioSession.CategoryOptions = []
+
+        let isAnyPlayerPlaying = videoViews.allObjects.contains { view in
+            view._player?.isMuted == false || (view._player != nil && view._player?.rate != 0)
+        }
+
+        let anyPlayerShowNotificationControls = videoViews.allObjects.contains { view in
+            view._showNotificationControls
+        }
+
+        let anyPlayerNeedsPiP = videoViews.allObjects.contains { view in
+            view.isPipEnabled()
+        }
+
+        let anyPlayerNeedsBackgroundPlayback = videoViews.allObjects.contains { view in
+            view._playInBackground
+        }
+
+        let canAllowMixing = !anyPlayerShowNotificationControls && !anyPlayerNeedsBackgroundPlayback
+
+        if canAllowMixing {
+            let shouldEnableMixing = videoViews.allObjects.contains { view in
+                view._mixWithOthers == "mix"
+            }
+
+            let shouldEnableDucking = videoViews.allObjects.contains { view in
+                view._mixWithOthers == "duck"
+            }
+
```

Comment:
I know in the original code theres no case for switching back to inherit, would it be possible to add it in ? 
Because once you set it to 'mix' or 'duck' theres no way to set it back to 'inherit'